File: Colleen/Colleen.c
/*
    Colleen.c
*/
#define CODE "/*%c    Colleen.c%c*/%c#define CODE %c%s%c%c#include <stdio.h>%c%cstatic void print() {%c	printf(CODE, 10, 10, 10, 34, CODE, 34, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10);%c}%c%cint main() {%c/*%c    this is the tricky part%c*/%c	print();%c	return 0;%c}%c"
#include <stdio.h>

static void print() {
	printf(CODE, 10, 10, 10, 34, CODE, 34, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10);
}

int main() {
/*
    this is the tricky part
*/
	print();
	return 0;
}

File: PY/Colleen/Colleen.py
#!/usr/bin/python

#
#   Colleen.py
#

CODE='#!/usr/bin/python%c%c#%c#   Colleen.py%c#%c%cCODE=%c%s%c%c%cdef print_it():%c    print(CODE %c (10, 10, 10, 10, 10, 10, 39, CODE, 39, 10, 10, 10, 37, 10, 10, 10, 10, 10, 10, 10, 10))%c%cdef main():%c#%c#   this is the tricky part%c#%c    print_it()%c%cmain()'

def print_it():
    print(CODE % (10, 10, 10, 10, 10, 10, 39, CODE, 39, 10, 10, 10, 37, 10, 10, 10, 10, 10, 10, 10, 10))

def main():
#
#   this is the tricky part
#
    print_it()

main()
